### PR TITLE
Replaced Toast with Snackbar in AudioView.kt & CustomStudyDialog.kt

### PR DESCRIPTION
**Pull request by Giyutomioka-SS**
## Purpose / Description
Replaced toast with Snackbar in AudioView.kt & CustomStudyDialog.kt

## Fixes
Related to #13411 

## How Has This Been Tested?
Tested on Realme 6

imgonline-com-ua-twotoone-VJeuRZul8Mwk

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by SanjaySargam — APPROVED** [review]
LGTM!!!

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt [hunk 1]
```diff
@@ -445,7 +446,7 @@ class CustomStudyDialog(private val collection: Collection, private val customSt
             Timber.i("Found deck: '%s'", customStudyDeck)
             if (cur.isStd) {
                 Timber.w("Deck: '%s' was non-dynamic", customStudyDeck)
-                showThemedToast(requireContext(), getString(R.string.custom_study_deck_exists), true)
+                showSnackbar(getString(R.string.custom_study_deck_exists), Snackbar.LENGTH_SHORT)
```
Comment (oakkitten) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt:449]: You don't have to call `getString` for simple strings. Didn't run this change but I think it should work

```suggestion
                showSnackbar(R.string.custom_study_deck_exists, Snackbar.LENGTH_SHORT)
```

Same in other places

**Review by oakkitten — COMMENTED** [review]
Would be nice to have at least one screenshot to make sure things are positioned correctly

**Comment by Giyutomioka-SS** [comment]
photo_2023-04-08_00-32-09
photo_2023-04-08_00-30-28

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt [hunk 2]
```diff
@@ -470,7 +471,7 @@ class CustomStudyDialog(private val collection: Collection, private val customSt
             // issue
             Timber.w("Invalid Dynamic Deck: %s", dyn)
             CrashReportService.sendExceptionReport("Custom Study Deck had no terms", "CustomStudyDialog - createCustomStudySession")
-            showThemedToast(requireContext(), getString(R.string.custom_study_rebuild_deck_corrupt), false)
+            showSnackbar((R.string.custom_study_rebuild_deck_corrupt))
```
Comment (oakkitten) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt:474]: Don't need double parentheses here

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioView.kt [hunk 3]
```diff
@@ -277,11 +279,7 @@ class AudioView private constructor(context: Context, resPlay: Int, resPause: In
                 // We can get to this screen without permissions through the "Pronunciation" feature.
                 if (!canRecordAudio(mContext)) {
                     Timber.w("Audio recording permission denied.")
-                    showThemedToast(
-                        mContext,
-                        resources.getString(R.string.multimedia_editor_audio_permission_denied),
-                        true
-                    )
+                    showSnackbar(resources.getString(R.string.multimedia_editor_audio_permission_denied), Snackbar.LENGTH_SHORT)
```
Comment (oakkitten) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioView.kt:282]: I think we can ditch the call to `resources.getString()` here as well

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioView.kt [hunk 4]
```diff
@@ -291,7 +289,7 @@ class AudioView private constructor(context: Context, resPlay: Int, resPause: In
                         } catch (e: Exception) {
                             // either output file failed or codec didn't work, in any case fail out
                             Timber.e("RecordButton.onClick() :: error recording to %s\n%s", audioPath, e.message)
-                            showThemedToast(mContext, gtxt(R.string.multimedia_editor_audio_view_recording_failed), true)
+                            showSnackbar(gtxt(R.string.multimedia_editor_audio_view_recording_failed), Snackbar.LENGTH_SHORT)
```
Comment (oakkitten) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioView.kt:292]: Looking a bit deeper, we should actually be able to get rid of the call to `gtxt` here and elsewhere, and also of the [method itself](https://github.com/ankidroid/Anki-Android/blob/3b3a2eab89ed07fa4c2330c7fdcc3bb5f9728d1f/AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioView.kt#L70-L72). A view has access to its resources, and we have a `View.showSnackbar()` method that takes an `Int` text resource.

**Review by oakkitten — CHANGES_REQUESTED** [review]
Thanks for the screenshots! Not using gesture navigation hint eh? Me too.

A few more simplifications to polish this off

**Review by oakkitten — APPROVED** [review]
LGTM 👍 (Disclaimer: didn't run the code)

**Comment by lukstbit** [comment]
This took some conflicts...

**Pull request closed without merging**